Add the commit SHA tag during building of precompiled image

shivakunv opened the pull request.

shivakunv commented on .github/workflows/precompiled.yaml:
```diff
@@ -15,9 +15,21 @@
+      - opened
+      - synchronize
+    branches:
+      - tagcommittshaimage
```
test code

shivakunv commented on .github/workflows/precompiled.yaml:
```diff
@@ -37,17 +49,19 @@ jobs:
         run: |
           # get driver_branch
-          DRIVER_BRANCH=("535" "550")
+          DRIVER_BRANCH=("535")
```
test code

shivakunv commented on .github/workflows/precompiled.yaml:
```diff
@@ -37,17 +49,19 @@ jobs:
           # get kernel flavors
-          KERNEL_FLAVORS=("aws" "azure" "generic" "nvidia" "oracle")
+          # KERNEL_FLAVORS=("aws" "azure" "generic" "nvidia" "oracle")
+          KERNEL_FLAVORS=("generic")
```
test code

shivakunv commented on .github/workflows/precompiled.yaml:
```diff
@@ -37,17 +49,19 @@ jobs:
           # get ubuntu distributions
-          DIST=("ubuntu22.04" "ubuntu24.04")
+          # DIST=("ubuntu22.04" "ubuntu24.04")
+          DIST=("ubuntu22.04")
```
test code

shivakunv commented on Makefile:
```diff
@@ -90,11 +90,11 @@ pull-signed_ubuntu20.04%: DRIVER_TAG = $(DRIVER_BRANCH)
 pull-signed_ubuntu22.04%: DIST = ubuntu22.04
 pull-signed_ubuntu22.04%: DRIVER_TAG = $(DRIVER_BRANCH)
-pull-signed_ubuntu22.04%: IMAGE_TAG = $(DRIVER_BRANCH)-$(KERNEL_VERSION)-$(DIST)
+pull-signed_ubuntu22.04%: IMAGE_TAG := $(if $(VERSION),$(VERSION)-)$(DRIVER_BRANCH)-$(KERNEL_VERSION)-$(DIST)
```
IMAGE_TAG = does lazy initiliazation
added := to expand the variable IMAGE_TAG :=  

if VERSION is non empty add VERSION- (commitsha-)

shivakunv commented on .github/workflows/precompiled.yaml:
```diff
@@ -389,6 +403,7 @@ jobs:
           retention-days: 15
 
   publish-precompiled-image:
+    if: "!github.event.COMMIT_SHORT_SHA"
```
do not publish in case of dev env

shivakunv commented on Makefile:
```diff
@@ -90,11 +90,11 @@ pull-signed_ubuntu20.04%: DRIVER_TAG = $(DRIVER_BRANCH)
 pull-signed_ubuntu22.04%: DIST = ubuntu22.04
 pull-signed_ubuntu22.04%: DRIVER_TAG = $(DRIVER_BRANCH)
-pull-signed_ubuntu22.04%: IMAGE_TAG = $(DRIVER_BRANCH)-$(KERNEL_VERSION)-$(DIST)
+pull-signed_ubuntu22.04%: IMAGE_TAG = $(if $(VERSION),$(VERSION)-)$(DRIVER_BRANCH)-$(KERNEL_VERSION)-$(DIST)
```
if VERSION is non empty add VERSION- (commitsha-)

shivakunv commented on .github/workflows/precompiled.yaml:
```diff
@@ -394,6 +405,7 @@ jobs:
       - set-driver-version-matrix
       - collect-e2e-test-matrix
       - e2e-tests-nvidiadriver
+    if: ${{ needs.collect-e2e-test-matrix.outputs.skip_publishing_image == '0' }}
```
skipping publishing of image in case of dev env

cdesiniotis commented:
As discussed during our call, I believe we need to remove the `VERSION` variable in `precompiled.yml` so that the commit sha is not also added to images built + pushed in our scheduled pipelines.
https://github.com/NVIDIA/gpu-driver-container/blob/da308fd0ecd8842c203399ca7f993e3b214cc216/.github/workflows/precompiled.yaml#L106
https://github.com/NVIDIA/gpu-driver-container/blob/da308fd0ecd8842c203399ca7f993e3b214cc216/.github/workflows/precompiled.yaml#L122

shivakunv commented:
> As discussed during our call, I believe we need to remove the `VERSION` variable in `precompiled.yml` so that the commit sha is not also added to images built + pushed in our scheduled pipelines.
> 
> https://github.com/NVIDIA/gpu-driver-container/blob/da308fd0ecd8842c203399ca7f993e3b214cc216/.github/workflows/precompiled.yaml#L106
> 
> 
> https://github.com/NVIDIA/gpu-driver-container/blob/da308fd0ecd8842c203399ca7f993e3b214cc216/.github/workflows/precompiled.yaml#L122
> 
> I may be mistaken though. I just want to make sure the image pushed to ghcr.io from our daily pipelines do not contain the SHA.

we are not using VERSION. removed

shivakunv commented:
> As discussed during our call, I believe we need to remove the `VERSION` variable in `precompiled.yml` so that the commit sha is not also added to images built + pushed in our scheduled pipelines.
> 
> https://github.com/NVIDIA/gpu-driver-container/blob/da308fd0ecd8842c203399ca7f993e3b214cc216/.github/workflows/precompiled.yaml#L106
> 
> 
> https://github.com/NVIDIA/gpu-driver-container/blob/da308fd0ecd8842c203399ca7f993e3b214cc216/.github/workflows/precompiled.yaml#L122
> 
> I may be mistaken though. I just want to make sure the image pushed to ghcr.io from our daily pipelines do not contain the SHA.

@cdesiniotis 
we are not using VERSION and COMMIT_SHORT_SHA . 
removed from precompile.yaml
